### PR TITLE
Improve LinkDetector

### PR DESCRIPTION
**Pull request by dgozman**
@isidorn Could you please take a look? This implements some improvements. We can try to unify with terminal link handling, however it doesn't seem to be straightforward because the control flow  is very different.

- recognizes web urls;
- checks that file path exists;
- supports relative to workspace root paths (./foo);
- supports relative to home dir paths (~/foo);
- optional line/column;
- smoke test.

### CODE REVIEW

File: src/vs/workbench/contrib/debug/common/replModel.ts [hunk 1]
```diff
@@ -119,12 +120,12 @@ export class ReplModel {
 		this.addReplElement(result);
 	}
 
-	appendToRepl(data: string | IExpression, sev: severity, source?: IReplElementSource): void {
+	appendToRepl(session: IDebugSession | undefined, data: string | IExpression, sev: severity, source?: IReplElementSource): void {
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/common/replModel.ts]: Here the session should always be defined, it should never be undefined (I think that case can not happen)

File: src/vs/workbench/contrib/debug/common/replModel.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ let topReplElementCounter = 0;
 
 export class SimpleReplElement implements IReplElement {
 	constructor(
+		public session: IDebugSession | undefined,
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/common/replModel.ts]: Session should never be undefined (same as below comment)

**Review by isidorn — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/debug/browser/linkDetector.ts [hunk 3]
```diff
@@ -3,160 +3,201 @@
...
-	private static FILE_LOCATION_PATTERNS: RegExp[] = [
-		// group 0: full path with line and column
-		// group 1: full path without line and column, matched by `*.*` in the end to work only on paths with extensions in the end (s.t. node:10352 would not match)
-		// group 2: drive letter on windows with trailing backslash or leading slash on mac/linux
-		// group 3: line number, matched by (:(\d+))
-		// group 4: column number, matched by ((?::(\d+))?)
-		// e.g.: at Context.<anonymous> (c:\Users\someone\Desktop\mocha-runner\test\test.js:26:11)
-		/(?![\(])(?:file:\/\/)?((?:([a-zA-Z]+:)|[^\(\)<>\'\"\[\]:\s]+)(?:[\\/][^\(\)<>\'\"\[\]:]*)?\.[a-zA-Z]+[0-9]*):(\d+)(?::(\d+))?/g
-	];
+const osPath = platform.isWindows ? win32 : posix;
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/linkDetector.ts]: I do not think `osPath` is needed here. You should just be able to use `path` and it should automatically use the correct one. Check out the `path` module

**Review by dgozman — COMMENTED** [review]
Thank you, this sounds good to me. I did address the comments.

There was a small change in behavior - we don't wrap text with extra `span` when not required. From manual testing, I didn't find any issues with that.

**Comment by isidorn** [comment]
Great, thanks for addressing the comments. Merging this in so we start self hosting on it already.
Thanks again for this contribution!

**Pull request merged**